imap/global.c

```diff
@@ -93,7 +93,7 @@ static enum {
 
 static int cyrus_init_nodb = 0;
 
-EXPORTED int in_shutdown = 0;
+EXPORTED volatile sig_atomic_t in_shutdown = 0;
```

Comment:
This change isn't actually needed anymore now that you've made cyr_expire just use lib/signals 😅